core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFile.java

```diff
@@ -120,6 +129,30 @@ public Map<String, Profile> profiles() {
         return profileMap != null ? Collections.unmodifiableMap(profileMap) : profileMap;
     }
 
+    /**
+     * Checks if the profile files have been changed on disk.
+     * This method will always return {@link Boolean#FALSE} if the {@link ProfileFile} instance was built
+     * using {@link Builder#content(InputStream)} since a stream cannot be re-run
+     * or checked if it's been updated.
+     *
+     * @return True if any files used to build this object have been modified.
+     * @see #isStaleAsOf(Instant)
+     */
+    public boolean isStale() {
+        return isStaleIfAny(diskFileWasModifiedSinceLoad());
+    }
+
+    /**
+     * Checks if the profile files have been changed on disk as of a particular instant.
+     *
+     * @param instant The time to check against for any file modifications.
+     * @return True if any files used to build this object have been modified.
+     * @see #isStale()
+     */
+    public boolean isStaleAsOf(Instant instant) {
+        return isStaleIfAny(details -> details.wasBuiltBefore(instant));
+    }
```

Comment:
Can these just be properties or internal details on the refresher class? IMO `ProfileFile` shouldn't have any concept of staleness. It should remain a relatively simple class that just contains `Profile`s and that's it.

Reply:
Moved this logic to `ProfileFileRefresher' and new class `ReloadingProfileCredentialsProvider`